File: apps/docs/src/code/python/timeout/timeout_filesystem.py
from e2b import Session

session = Session.create(id="Nodejs")

# Timeout 3s for the write operation
session.filesystem.write("test.txt", "Hello World", timeout=3) # $HighlightLine

# Timeout 3s for the list operation
dir_content = session.filesystem.list(".", timeout=3) # $HighlightLine
print(dir_content)

# Timeout 3s for the read operation
file_content = session.filesystem.read("test.txt", timeout=3) # $HighlightLine
print(file_content)

session.close()

File: apps/docs/src/code/js/timeout/timeout_filesystem.js
import { Session } from '@e2b/sdk'

const session = await Session.create({ id: 'Nodejs' })

// Timeout 3s for the write operation
await session.filesystem.write('hello.txt', 'Hello World!', { timeout: 3000 }) // $HighlightLine

// Timeout 3s for the list operation
const files = await session.filesystem.list('.', { timeout: 3000 }) // $HighlightLine
console.log(files)

// Timeout 3s for the read operation
const content = await session.filesystem.read('hello.txt', { timeout: 3000 }) // $HighlightLine
console.log(content)

await session.close()
